Cargo.toml

```diff
@@ -0,0 +1,16 @@
+[package]
+name = "linux-loader"
+version = "0.1.0"
+authors = ["cathy <cathy.zhang@intel.com>"]
+edition = "2018"
+license = "Apache-2.0 AND BSD-3-Clause"
+
+[features]
+default = []
+elf = []
+bzImage = []
+
+
+[dependencies.vm-memory]
+git = "https://github.com/rust-vmm/vm-memory"
```

Comment:
I think the dependency should be locked onto some specific revision, otherwise there is no way to tell when the crate will be broken by a change on vm-memory.

Reply:
I think it's sensible to point to rust-vmm/vm-memory branch. There are several PRs for vm-memory under review, once they are merged into master, we will fix it if the crate is broken by the dependency at that time. :-)

Reply:
Well that's the point, to make sure this crate don't get broken by changes in vm-memory.
Right now, this crate works with the latest version of vm-memory, and we want this to continue to be true.
Once a new PR from vm-memory gets merged, if we want linux-loader to rely on the latest version, then this means we will need to submit a new PR to linux-loader to update the revision from vm-memory and potentially fix the code.

Reply:
@sboeuf @rbradford @sameo 

I'm sincerely sorry, I just have chance to go through what is exactly done in the vendor PR sebastien create, linux-loader with 2b95f1e1958a2b6399b590f64344cab5b4a6d608 is pinned in cloud-hypervisor for use. But actually, it does not appear since I did git reset during recently changes. it broke the build for cloud-hypervisor, I'd like to update Cargo.toml with new existing commit, but seems lock file under .cargo need to  be removed firstly for new generation, which I could not touch directly. so I need your help to fix that. Really sorry for the inconvenience, I will pay attention on using git reset later.

Meanwhile, with the commit that remove dependency of GuestMemoryMmap, there is change required from cloud-hypervisor to pass GuestMemory to loader also. I could see how to change that if you need once commit get update in cloud-hypervisor.

Reply:
I don't think we need vendoring. Push forces are not allowed on the master branch of rust-vmm/vm-memory. What we do need is a fixed revision. We can either use a commit sha1 here or add tags to vm-memory and use tags in dependencies.

Reply:
@sameo @bjzhjing do you want to use a fixed revision in this PR or do you want to fix it in a future PR?